source/MaterialXRuntime/Private/PvtPort.h

```diff
@@ -71,72 +71,21 @@ class PvtPort : public PvtObject
         return (_flags & RtPortFlag::SOCKET) != 0;
     }
 
-    const RtIdentifier& getColorSpace() const
-    {
-        const RtTypedValue* attr = getAttribute(PvtPort::COLOR_SPACE, RtType::IDENTIFIER);
-        return attr ? attr->asIdentifier() : EMPTY_IDENTIFIER;
-    }
-
-    void setColorSpace(const RtIdentifier& colorspace)
-    {
-        RtTypedValue* attr = createAttribute(PvtPort::COLOR_SPACE, RtType::IDENTIFIER);
-        attr->asIdentifier() = colorspace;
-    }
-
-    const RtIdentifier& getUnit() const
-    {
-        const RtTypedValue* attr = getAttribute(PvtPort::UNIT, RtType::IDENTIFIER);
-        return attr ? attr->asIdentifier() : EMPTY_IDENTIFIER;
-    }
-
-    void setUnit(const RtIdentifier& unit)
-    {
-        RtTypedValue* attr = createAttribute(PvtPort::UNIT, RtType::IDENTIFIER);
-        attr->asIdentifier() = unit;
-    }
-
-    const RtIdentifier& getUnitType() const
-    {
-        const RtTypedValue* attr = getAttribute(PvtPort::UNIT_TYPE, RtType::IDENTIFIER);
-        return attr ? attr->asIdentifier() : EMPTY_IDENTIFIER;
-    }
-
-    void setUnitType(const RtIdentifier& unit)
+    bool isUniform() const
     {
-        RtTypedValue* attr = createAttribute(PvtPort::UNIT_TYPE, RtType::IDENTIFIER);
-        attr->asIdentifier() = unit;
+        return (_flags & RtPortFlag::UNIFORM) != 0;
     }
 
     bool isToken() const
     {
         return (_flags & RtPortFlag::TOKEN) != 0;
     }
 
-    void setIsToken(bool val)
```

Comment:
Removed setIsToken as well as setUniform below. Once a port has been created these flags should not be editable.